web/static/js/components/Layout.js

```diff
@@ -29,7 +29,9 @@ class Layout extends React.Component {
           </nav>
         </header>
 
-        <NotificationList />
+        {Auth.isLoggedIn() &&
+          <NotificationList />
+        }
```

Comment:
We can't connect to the notifications channel without a token, so we shouldn't try if we're not logged in.